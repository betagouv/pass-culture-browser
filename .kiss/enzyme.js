import React from 'react'
import { shallow } from 'enzyme'

import MyComponent from '../MyComponent'

describe('src | components | MyComponent', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      const props = { errors: {} }
      const wrapper = shallow(<MyComponent {...props} />)
      expect(wrapper).toBeDefined()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
