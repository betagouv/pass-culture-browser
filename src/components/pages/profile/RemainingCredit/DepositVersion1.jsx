import PropTypes from 'prop-types'
import React, { PureComponent } from 'react'

import { NON_BREAKING_SPACE } from '../../../../utils/specialCharacters'
import { getRemainingCreditForGivenCreditLimit } from '../domain/getRemainingCreditForGivenCreditLimit'
import CreditGauge from './CreditGauge/CreditGauge'
import User from '../ValueObjects/User'

class DepositVersion1 extends PureComponent {
  constructor(props) {
    super(props)
    this.state = {
      isReadMoreVisible: false,
    }
  }

  handleToggleReadMore = () => {
    this.setState(previousState => ({ isReadMoreVisible: !previousState.isReadMoreVisible }))
  }

  render() {
    const { user } = this.props
    const { isReadMoreVisible } = this.state
    const { expenses, wallet_balance: walletBalance } = user

    let initialDeposit = 0
    if (expenses.length) {
      const all = expenses.find(expense => expense.domain === 'all')
      initialDeposit = all.limit
    }

    const totalGaugeProperties = {
      creditLimit: initialDeposit,
      remainingCredit: walletBalance,
    }
    let physicalCreditLimit = 0
    let physicalRemainingCredit = 0
    if (expenses.length && user.deposit_version === 1) {
      const physical = expenses.find(expense => expense.domain === 'physical')
      physicalCreditLimit = physical.limit
      physicalRemainingCredit = getRemainingCreditForGivenCreditLimit(walletBalance)(physical)
    }

    const physicalGaugeProperties = {
      creditLimit: physicalCreditLimit,
      remainingCredit: physicalRemainingCredit,
    }
    let digitalCreditLimit = 0
    let digitalRemainingCredit = 0
    if (expenses.length) {
      const digital = expenses.find(expense => expense.domain === 'digital')
      digitalCreditLimit = digital.limit
      digitalRemainingCredit = getRemainingCreditForGivenCreditLimit(walletBalance)(digital)
    }

    const digitalGaugeProperties = {
      creditLimit: digitalCreditLimit,
      remainingCredit: digitalRemainingCredit,
    }

    return (
      <section className="pf-section">
        <div className="rc-informations-container">
          <div className="rc-gauges-container">
            <div className="rc-gauges-title">
              {`Tu peux encore dépenser jusqu’à${NON_BREAKING_SPACE}:`}
            </div>
            <div className="rc-gauges">
              <CreditGauge
                extraClassName="gauge-digital"
                picto="picto-digital-good"
                {...digitalGaugeProperties}
              >
                {`en offres\u000Anumériques\u000A(streaming…)`}
              </CreditGauge>
              <CreditGauge
                extraClassName="gauge-physical"
                picto="picto-physical-good"
                {...physicalGaugeProperties}
              >
                {`en offres\u000Aphysiques\u000A(livres…)`}
              </CreditGauge>
              <CreditGauge
                extraClassName="gauge-total"
                picto="picto-ticket"
                {...totalGaugeProperties}
              >
                {`en sorties\u000A(spectacles…)`}
              </CreditGauge>
            </div>
          </div>
          <div className="rc-read-more">
            <button
              className={`rc-read-more-button ${
                isReadMoreVisible ? 'rc-read-more-drop-down' : 'rc-read-more-drop-down-flipped'
              }`}
              onClick={this.handleToggleReadMore}
              type="button"
            >
              {`Pourquoi les biens physiques et numériques sont-ils limités${NON_BREAKING_SPACE}?`}
            </button>
            {isReadMoreVisible && (
              <p className="rc-read-more-content">
                {`Le but du pass Culture est de renforcer tes pratiques culturelles,
                mais aussi d’en créer de nouvelles. Ces plafonds ont été mis en place
                pour favoriser la diversification des pratiques culturelles.`}
              </p>
            )}
          </div>
        </div>
      </section>
    )
  }
}

DepositVersion1.propTypes = {
  user: PropTypes.shape(User).isRequired,
}

export default DepositVersion1
