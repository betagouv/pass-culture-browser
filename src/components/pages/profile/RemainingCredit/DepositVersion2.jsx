import PropTypes from 'prop-types'
import React, { PureComponent } from 'react'

import { NON_BREAKING_SPACE } from '../../../../utils/specialCharacters'
import { getRemainingCreditForGivenCreditLimit } from '../domain/getRemainingCreditForGivenCreditLimit'
import CreditGauge from './CreditGauge/CreditGauge'
import User from '../ValueObjects/User'

class DepositVersion2 extends PureComponent {
  constructor(props) {
    super(props)
    this.state = {
      isReadMoreVisible: false,
    }
  }

  handleToggleReadMore = () => {
    this.setState(previousState => ({ isReadMoreVisible: !previousState.isReadMoreVisible }))
  }

  render() {
    const { user } = this.props
    const { isReadMoreVisible } = this.state
    const { expenses, wallet_balance: walletBalance } = user

    // Gauge ALL
    let initialDeposit = 0
    if (expenses.length) {
      const all = expenses.find(expense => expense.domain === 'all')
      initialDeposit = all.limit
    }

    const totalGaugeProperties = {
      creditLimit: initialDeposit,
      remainingCredit: walletBalance,
    }
    // Gauge ALL

    let digitalCreditLimit = 0
    let digitalRemainingCredit = 0
    if (expenses.length) {
      const digital = expenses.find(expense => expense.domain === 'digital')
      digitalCreditLimit = digital.limit
      digitalRemainingCredit = getRemainingCreditForGivenCreditLimit(walletBalance)(digital)
    }

    const digitalGaugeProperties = {
      creditLimit: digitalCreditLimit,
      remainingCredit: digitalRemainingCredit,
    }

    return (
      <section className="pf-section">
        <div className="rc-informations-container">
          <div className="rc-gauges-container">
            <div className="rc-gauges-title">
              {`Tu peux encore dépenser jusqu’à${NON_BREAKING_SPACE}:`}
            </div>
            <div className="rc-gauges">
              <CreditGauge
                extraClassName="gauge-digital"
                picto="picto-digital-good"
                {...digitalGaugeProperties}
              >
                {`en offres\u000Anumériques\u000A(streaming…)`}
              </CreditGauge>
              <CreditGauge
                extraClassName="gauge-total"
                picto="picto-all-goods"
                {...totalGaugeProperties}
              >
                {`en sorties et\u000Abiens physiques\u000A(concerts, livres…)`}
              </CreditGauge>
            </div>
          </div>
          <div className="rc-read-more">
            <button
              className={`rc-read-more-button ${
                isReadMoreVisible ? 'rc-read-more-drop-down' : 'rc-read-more-drop-down-flipped'
              }`}
              onClick={this.handleToggleReadMore}
              type="button"
            >
              {`Pourquoi les biens numériques sont-ils limités${NON_BREAKING_SPACE}?`}
            </button>
            {isReadMoreVisible && (
              <p className="rc-read-more-content">
                {`Le but du pass Culture est de renforcer tes pratiques culturelles,
                mais aussi d’en créer de nouvelles. Ces plafonds ont été mis en place
                pour favoriser la diversification des pratiques culturelles.`}
              </p>
            )}
          </div>
        </div>
      </section>
    )
  }
}

DepositVersion2.propTypes = {
  user: PropTypes.shape(User).isRequired,
}

export default DepositVersion2
