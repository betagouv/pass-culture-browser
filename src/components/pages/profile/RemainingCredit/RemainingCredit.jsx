import PropTypes from 'prop-types'
import React from 'react'

import getAvailableBalanceByType from '../utils/utils'
import getWalletValue from '../../../../utils/user/getWalletValue'
import { formatEndValidityDate } from '../../../../utils/date/date'

const RemainingCredit = ({ currentUser }) => {
  const { expenses, wallet_date_created } = currentUser || {}
  const allWallet = getWalletValue(currentUser)
  const { digital, physical } = expenses
  const [digitalAvailable, physicalAvailable] = [digital, physical].map(
    getAvailableBalanceByType(allWallet)
  )
  let endValidityDate = null
  if (wallet_date_created) {
    endValidityDate = formatEndValidityDate(new Date(wallet_date_created))
  }

  return (
    <div id="profile-page-remaining-credit">
      <h3 className="dotted-bottom-primary pb8 px12 is-italic is-uppercase is-primary-text fs15 is-normal">
        {'Crédit restant'}
      </h3>
      <div className="mt12 px12">
        <div
          className="jauges py12 text-center"
          id="profile-page-user-wallet"
        >
          <div className="text overall">
            <b
              className="is-block"
              id="profile-wallet-balance-value"
            >
              {`Il reste ${allWallet} €`}
            </b>
            <span className="is-block fs14">
              {'sur votre pass Culture'}
            </span>
          </div>
          <div className="text-containers mt12 py12 mr8">
            <div
              className="fs15"
              id="profile-physical-wallet-value"
            >
              {'Jusqu’à '}
              <b>
                {`${physicalAvailable} €`}
              </b>
              {' pour les biens culturels'}
            </div>
            <div
              className="fs15 mt12"
              id="profile-digital-wallet-value"
            >
              {'Jusqu’à '}
              <b>
                {`${digitalAvailable} €`}
              </b>
              {' pour les offres numériques'}
            </div>
            {endValidityDate && (
              <div
                className="fs15 mt12"
                id="profile-end-validity-date"
              >
                {`Votre crédit est valable jusqu’au ${endValidityDate}.`}
              </div>
            )}
          </div>
        </div>
      </div>
    </div>
  )
}

RemainingCredit.propTypes = {
  currentUser: PropTypes.shape().isRequired
}

export default RemainingCredit
