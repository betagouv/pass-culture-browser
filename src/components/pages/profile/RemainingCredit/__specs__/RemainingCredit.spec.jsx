import React from 'react'
import { shallow } from 'enzyme'

import RemainingCredit from '../RemainingCredit'

const walletId = '#profile-wallet-balance-value'
const digitalId = '#profile-digital-wallet-value'
const physicalId = '#profile-physical-wallet-value'
const endValidityDateId = '#profile-end-validity-date'

describe('components | RemainingCredit', () => {
  it('should match the snapshot with required props', () => {
    // given
    const props = {
      currentUser: {
        expenses: {
          digital: { actual: 0, max: 200 },
          physical: { actual: 0, max: 200 },
        },
        wallet_balance: 500,
        wallet_date_created: '2019-09-10T08:05:45.778894Z',
      },
    }

    // when
    const wrapper = shallow(<RemainingCredit {...props} />)

    // then
    expect(wrapper).toMatchSnapshot()
  })

  describe('render', () => {
    it('should include Crédit restant title', () => {
      // given
      const props = {
        currentUser: {
          expenses: {
            digital: { actual: 100, max: 200 },
            physical: { actual: 100, max: 200 },
          },
          wallet_balance: 0,
        },
      }

      // when
      const wrapper = shallow(<RemainingCredit {...props} />)

      // then
      const title = wrapper.find('h3').text()
      expect(title).toBe('Crédit restant')
    })

    it('with wallet value set to 0', () => {
      // given
      const props = {
        currentUser: {
          expenses: {
            digital: { actual: 100, max: 200 },
            physical: { actual: 100, max: 200 },
          },
          wallet_balance: 0,
        },
      }

      // when
      const wrapper = shallow(<RemainingCredit {...props} />)
      const walletElement = wrapper.find(walletId)
      const digitalElement = wrapper.find(digitalId)
      const physicalElement = wrapper.find(physicalId)

      // then
      let expected = `Il reste 0 €`
      let text = walletElement.text()
      expect(text).toBe(expected)

      expected = `Jusqu’à 0 € pour les offres numériques`
      text = digitalElement.text()
      expect(digitalElement.text()).toBe(expected)
      expected = `Jusqu’à 0 € pour les biens culturels`

      text = physicalElement.text()
      expect(physicalElement.text()).toBe(expected)
    })

    it('with wallet value set to 90', () => {
      // given
      const props = {
        currentUser: {
          expenses: {
            digital: { actual: 20, max: 200 },
            physical: { actual: 80, max: 200 },
          },
          wallet_balance: 90,
        },
      }

      // when
      const wrapper = shallow(<RemainingCredit {...props} />)
      const walletElement = wrapper.find(walletId)
      const digitalElement = wrapper.find(digitalId)
      const physicalElement = wrapper.find(physicalId)

      // then
      let expected = `Il reste 90 €`
      let text = walletElement.text()
      expect(text).toBe(expected)

      expected = `Jusqu’à 90 € pour les offres numériques`
      text = digitalElement.text()
      expect(digitalElement.text()).toBe(expected)
      expected = `Jusqu’à 90 € pour les biens culturels`

      text = physicalElement.text()
      expect(physicalElement.text()).toBe(expected)
    })

    it('render with wallet value set to 200', () => {
      // given
      const props = {
        currentUser: {
          expenses: {
            digital: { actual: 120, max: 200 },
            physical: { actual: 140, max: 200 },
          },
          wallet_balance: 200,
        },
      }

      // when
      const wrapper = shallow(<RemainingCredit {...props} />)
      const walletElement = wrapper.find(walletId)
      const digitalElement = wrapper.find(digitalId)
      const physicalElement = wrapper.find(physicalId)

      // then
      let expected = `Il reste 200 €`
      let text = walletElement.text()
      expect(text).toBe(expected)

      expected = `Jusqu’à 80 € pour les offres numériques`
      text = digitalElement.text()
      expect(digitalElement.text()).toBe(expected)

      expected = `Jusqu’à 60 € pour les biens culturels`
      text = physicalElement.text()
      expect(physicalElement.text()).toBe(expected)
    })

    it('should render end validity date', () => {
      // given
      const props = {
        currentUser: {
          expenses: {
            digital: { actual: 120, max: 200 },
            physical: { actual: 140, max: 200 },
          },
          wallet_balance: 90,
          wallet_date_created: '2019-09-10T08:05:45.778894Z',
        },
      }

      // when
      const wrapper = shallow(<RemainingCredit {...props} />)

      // then
      const textWithEndValidityDate = wrapper.find(endValidityDateId).text()
      expect(textWithEndValidityDate).toBe('Votre crédit est valable jusqu’au 2021 M09 10.')
    })

    it('should not render end validity date when user has no deposit', () => {
      // given
      const props = {
        currentUser: {
          expenses: {
            digital: { actual: 0, max: 200 },
            physical: { actual: 0, max: 200 },
          },
          wallet_balance: 0,
          wallet_date_created: null,
        },
      }

      // when
      const wrapper = shallow(<RemainingCredit {...props} />)
      const wrapperWithEndValidityDate = wrapper.find(endValidityDateId)

      // then
      expect(wrapperWithEndValidityDate).toHaveLength(0)
    })
  })
})
