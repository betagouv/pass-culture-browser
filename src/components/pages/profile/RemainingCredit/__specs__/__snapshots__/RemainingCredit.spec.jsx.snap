// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components | RemainingCredit should match the snapshot with required props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RemainingCredit
    currentUser={
      Object {
        "expenses": Object {
          "digital": Object {
            "actual": 0,
            "max": 200,
          },
          "physical": Object {
            "actual": 0,
            "max": 200,
          },
        },
        "wallet_balance": 500,
        "wallet_date_created": "2019-09-10T08:05:45.778894Z",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="dotted-bottom-primary pb8 px12 is-italic is-uppercase is-primary-text fs15 is-normal"
        >
          Crédit restant
        </h3>,
        <div
          className="mt12 px12"
        >
          <div
            className="jauges py12 text-center"
            id="profile-page-user-wallet"
          >
            <div
              className="text overall"
            >
              <b
                className="is-block"
                id="profile-wallet-balance-value"
              >
                Il reste 500 €
              </b>
              <span
                className="is-block fs14"
              >
                sur votre pass Culture
              </span>
            </div>
            <div
              className="text-containers mt12 py12 mr8"
            >
              <div
                className="fs15"
                id="profile-physical-wallet-value"
              >
                Jusqu’à 
                <b>
                  200 €
                </b>
                 pour les biens culturels
              </div>
              <div
                className="fs15 mt12"
                id="profile-digital-wallet-value"
              >
                Jusqu’à 
                <b>
                  200 €
                </b>
                 pour les offres numériques
              </div>
              <div
                className="fs15 mt12"
                id="profile-end-validity-date"
              >
                Votre crédit est valable jusqu’au 2020 M09 10.
              </div>
            </div>
          </div>
        </div>,
      ],
      "id": "profile-page-remaining-credit",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Crédit restant",
          "className": "dotted-bottom-primary pb8 px12 is-italic is-uppercase is-primary-text fs15 is-normal",
        },
        "ref": null,
        "rendered": "Crédit restant",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="jauges py12 text-center"
            id="profile-page-user-wallet"
          >
            <div
              className="text overall"
            >
              <b
                className="is-block"
                id="profile-wallet-balance-value"
              >
                Il reste 500 €
              </b>
              <span
                className="is-block fs14"
              >
                sur votre pass Culture
              </span>
            </div>
            <div
              className="text-containers mt12 py12 mr8"
            >
              <div
                className="fs15"
                id="profile-physical-wallet-value"
              >
                Jusqu’à 
                <b>
                  200 €
                </b>
                 pour les biens culturels
              </div>
              <div
                className="fs15 mt12"
                id="profile-digital-wallet-value"
              >
                Jusqu’à 
                <b>
                  200 €
                </b>
                 pour les offres numériques
              </div>
              <div
                className="fs15 mt12"
                id="profile-end-validity-date"
              >
                Votre crédit est valable jusqu’au 2020 M09 10.
              </div>
            </div>
          </div>,
          "className": "mt12 px12",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="text overall"
              >
                <b
                  className="is-block"
                  id="profile-wallet-balance-value"
                >
                  Il reste 500 €
                </b>
                <span
                  className="is-block fs14"
                >
                  sur votre pass Culture
                </span>
              </div>,
              <div
                className="text-containers mt12 py12 mr8"
              >
                <div
                  className="fs15"
                  id="profile-physical-wallet-value"
                >
                  Jusqu’à 
                  <b>
                    200 €
                  </b>
                   pour les biens culturels
                </div>
                <div
                  className="fs15 mt12"
                  id="profile-digital-wallet-value"
                >
                  Jusqu’à 
                  <b>
                    200 €
                  </b>
                   pour les offres numériques
                </div>
                <div
                  className="fs15 mt12"
                  id="profile-end-validity-date"
                >
                  Votre crédit est valable jusqu’au 2020 M09 10.
                </div>
              </div>,
            ],
            "className": "jauges py12 text-center",
            "id": "profile-page-user-wallet",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <b
                    className="is-block"
                    id="profile-wallet-balance-value"
                  >
                    Il reste 500 €
                  </b>,
                  <span
                    className="is-block fs14"
                  >
                    sur votre pass Culture
                  </span>,
                ],
                "className": "text overall",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Il reste 500 €",
                    "className": "is-block",
                    "id": "profile-wallet-balance-value",
                  },
                  "ref": null,
                  "rendered": "Il reste 500 €",
                  "type": "b",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "sur votre pass Culture",
                    "className": "is-block fs14",
                  },
                  "ref": null,
                  "rendered": "sur votre pass Culture",
                  "type": "span",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="fs15"
                    id="profile-physical-wallet-value"
                  >
                    Jusqu’à 
                    <b>
                      200 €
                    </b>
                     pour les biens culturels
                  </div>,
                  <div
                    className="fs15 mt12"
                    id="profile-digital-wallet-value"
                  >
                    Jusqu’à 
                    <b>
                      200 €
                    </b>
                     pour les offres numériques
                  </div>,
                  <div
                    className="fs15 mt12"
                    id="profile-end-validity-date"
                  >
                    Votre crédit est valable jusqu’au 2020 M09 10.
                  </div>,
                ],
                "className": "text-containers mt12 py12 mr8",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Jusqu’à ",
                      <b>
                        200 €
                      </b>,
                      " pour les biens culturels",
                    ],
                    "className": "fs15",
                    "id": "profile-physical-wallet-value",
                  },
                  "ref": null,
                  "rendered": Array [
                    "Jusqu’à ",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "200 €",
                      },
                      "ref": null,
                      "rendered": "200 €",
                      "type": "b",
                    },
                    " pour les biens culturels",
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Jusqu’à ",
                      <b>
                        200 €
                      </b>,
                      " pour les offres numériques",
                    ],
                    "className": "fs15 mt12",
                    "id": "profile-digital-wallet-value",
                  },
                  "ref": null,
                  "rendered": Array [
                    "Jusqu’à ",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "200 €",
                      },
                      "ref": null,
                      "rendered": "200 €",
                      "type": "b",
                    },
                    " pour les offres numériques",
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Votre crédit est valable jusqu’au 2020 M09 10.",
                    "className": "fs15 mt12",
                    "id": "profile-end-validity-date",
                  },
                  "ref": null,
                  "rendered": "Votre crédit est valable jusqu’au 2020 M09 10.",
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="dotted-bottom-primary pb8 px12 is-italic is-uppercase is-primary-text fs15 is-normal"
          >
            Crédit restant
          </h3>,
          <div
            className="mt12 px12"
          >
            <div
              className="jauges py12 text-center"
              id="profile-page-user-wallet"
            >
              <div
                className="text overall"
              >
                <b
                  className="is-block"
                  id="profile-wallet-balance-value"
                >
                  Il reste 500 €
                </b>
                <span
                  className="is-block fs14"
                >
                  sur votre pass Culture
                </span>
              </div>
              <div
                className="text-containers mt12 py12 mr8"
              >
                <div
                  className="fs15"
                  id="profile-physical-wallet-value"
                >
                  Jusqu’à 
                  <b>
                    200 €
                  </b>
                   pour les biens culturels
                </div>
                <div
                  className="fs15 mt12"
                  id="profile-digital-wallet-value"
                >
                  Jusqu’à 
                  <b>
                    200 €
                  </b>
                   pour les offres numériques
                </div>
                <div
                  className="fs15 mt12"
                  id="profile-end-validity-date"
                >
                  Votre crédit est valable jusqu’au 2020 M09 10.
                </div>
              </div>
            </div>
          </div>,
        ],
        "id": "profile-page-remaining-credit",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Crédit restant",
            "className": "dotted-bottom-primary pb8 px12 is-italic is-uppercase is-primary-text fs15 is-normal",
          },
          "ref": null,
          "rendered": "Crédit restant",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="jauges py12 text-center"
              id="profile-page-user-wallet"
            >
              <div
                className="text overall"
              >
                <b
                  className="is-block"
                  id="profile-wallet-balance-value"
                >
                  Il reste 500 €
                </b>
                <span
                  className="is-block fs14"
                >
                  sur votre pass Culture
                </span>
              </div>
              <div
                className="text-containers mt12 py12 mr8"
              >
                <div
                  className="fs15"
                  id="profile-physical-wallet-value"
                >
                  Jusqu’à 
                  <b>
                    200 €
                  </b>
                   pour les biens culturels
                </div>
                <div
                  className="fs15 mt12"
                  id="profile-digital-wallet-value"
                >
                  Jusqu’à 
                  <b>
                    200 €
                  </b>
                   pour les offres numériques
                </div>
                <div
                  className="fs15 mt12"
                  id="profile-end-validity-date"
                >
                  Votre crédit est valable jusqu’au 2020 M09 10.
                </div>
              </div>
            </div>,
            "className": "mt12 px12",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="text overall"
                >
                  <b
                    className="is-block"
                    id="profile-wallet-balance-value"
                  >
                    Il reste 500 €
                  </b>
                  <span
                    className="is-block fs14"
                  >
                    sur votre pass Culture
                  </span>
                </div>,
                <div
                  className="text-containers mt12 py12 mr8"
                >
                  <div
                    className="fs15"
                    id="profile-physical-wallet-value"
                  >
                    Jusqu’à 
                    <b>
                      200 €
                    </b>
                     pour les biens culturels
                  </div>
                  <div
                    className="fs15 mt12"
                    id="profile-digital-wallet-value"
                  >
                    Jusqu’à 
                    <b>
                      200 €
                    </b>
                     pour les offres numériques
                  </div>
                  <div
                    className="fs15 mt12"
                    id="profile-end-validity-date"
                  >
                    Votre crédit est valable jusqu’au 2020 M09 10.
                  </div>
                </div>,
              ],
              "className": "jauges py12 text-center",
              "id": "profile-page-user-wallet",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <b
                      className="is-block"
                      id="profile-wallet-balance-value"
                    >
                      Il reste 500 €
                    </b>,
                    <span
                      className="is-block fs14"
                    >
                      sur votre pass Culture
                    </span>,
                  ],
                  "className": "text overall",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Il reste 500 €",
                      "className": "is-block",
                      "id": "profile-wallet-balance-value",
                    },
                    "ref": null,
                    "rendered": "Il reste 500 €",
                    "type": "b",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "sur votre pass Culture",
                      "className": "is-block fs14",
                    },
                    "ref": null,
                    "rendered": "sur votre pass Culture",
                    "type": "span",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div
                      className="fs15"
                      id="profile-physical-wallet-value"
                    >
                      Jusqu’à 
                      <b>
                        200 €
                      </b>
                       pour les biens culturels
                    </div>,
                    <div
                      className="fs15 mt12"
                      id="profile-digital-wallet-value"
                    >
                      Jusqu’à 
                      <b>
                        200 €
                      </b>
                       pour les offres numériques
                    </div>,
                    <div
                      className="fs15 mt12"
                      id="profile-end-validity-date"
                    >
                      Votre crédit est valable jusqu’au 2020 M09 10.
                    </div>,
                  ],
                  "className": "text-containers mt12 py12 mr8",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "Jusqu’à ",
                        <b>
                          200 €
                        </b>,
                        " pour les biens culturels",
                      ],
                      "className": "fs15",
                      "id": "profile-physical-wallet-value",
                    },
                    "ref": null,
                    "rendered": Array [
                      "Jusqu’à ",
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "200 €",
                        },
                        "ref": null,
                        "rendered": "200 €",
                        "type": "b",
                      },
                      " pour les biens culturels",
                    ],
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "Jusqu’à ",
                        <b>
                          200 €
                        </b>,
                        " pour les offres numériques",
                      ],
                      "className": "fs15 mt12",
                      "id": "profile-digital-wallet-value",
                    },
                    "ref": null,
                    "rendered": Array [
                      "Jusqu’à ",
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "200 €",
                        },
                        "ref": null,
                        "rendered": "200 €",
                        "type": "b",
                      },
                      " pour les offres numériques",
                    ],
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Votre crédit est valable jusqu’au 2020 M09 10.",
                      "className": "fs15 mt12",
                      "id": "profile-end-validity-date",
                    },
                    "ref": null,
                    "rendered": "Votre crédit est valable jusqu’au 2020 M09 10.",
                    "type": "div",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
